### PR TITLE
fix(cli): Support circular dependency imports

### PR DESCRIPTION
**Pull request by tchock**

### CODE REVIEW

File: packages/cli/src/steps/compile/runWebpackCompiler.ts [hunk 1]
```diff
@@ -82,12 +82,17 @@ function getFilePath(resource = '') {
...
 }
 
-function addDependencies(dependencyPackages) {
+function addDependencies(dependencyPackages, existingDependencies: string[] = []) {
   return (module): Module => {
     const isExternal = !!module.external;
     const isNodeModule = checkNodeModule(module.resource);
     const packageName = isNodeModule || isExternal ? getPackageName(module) : undefined;
     const filePath = module.resource && getFilePath(module.resource);
+    const isCircularImport = !!existingDependencies.find(
```
Comment (JohannesMerz) [inline packages/cli/src/steps/compile/runWebpackCompiler.ts:91]: I would have used a hash map here as lookup would be much faster than finding sth in an array

**Review by JohannesMerz — CHANGES_REQUESTED** [review]
Maybe sth to consider

File: packages/cli/src/steps/compile/runWebpackCompiler.ts (hunk 1 above)
Comment (tchock) [reply packages/cli/src/steps/compile/runWebpackCompiler.ts:91]: Thought about that as well. Let's use a map :)

Comment (tchock) [reply packages/cli/src/steps/compile/runWebpackCompiler.ts]: I used a `Set`, as we only need the path and it also has a `has()` function :)

**Review by JohannesMerz — APPROVED** [review]

**Pull request merged**